app/src/features/projects/view/components/TeamMember.tsx

```diff
@@ -0,0 +1,100 @@
+import { mdiAccountEdit, mdiCrown } from '@mdi/js';
+import Icon from '@mdi/react';
+import { Box, Stack, Typography } from '@mui/material';
+import assert from 'assert';
+import { ProjectContext } from 'contexts/projectContext';
+import { useContext, useMemo } from 'react';
+
+interface IProjectParticipantsRoles {
+  display_name: string;
+  roles: string[];
+  initials: string;
+  avatarColor: string;
+}
+
+const TeamMembers = () => {
+  const projectContext = useContext(ProjectContext);
+
+  // Project data must be loaded by a parent before this component is rendered
+  assert(projectContext.projectDataLoader.data);
+
+  // Define a custom sorting order for roles
+  const roleOrder: { [key: string]: number } = {
+    Coordinator: 1,
+    Collaborator: 2,
+    Observer: 3
+  };
+
+  const projectTeamMembers: IProjectParticipantsRoles[] = useMemo(
+    () =>
+      projectContext.projectDataLoader.data?.projectData.participants
+        .map((member) => {
+          const initials = member.display_name
+            .split(',')
+            .map((name) => name.trim().slice(0, 1).toUpperCase())
+            .reverse()
+            .join('');
+          return {
+            display_name: member.display_name,
+            roles: member.project_role_names,
+            avatarColor: getRandomHexColor(member.system_user_id),
+            initials
+          };
+        })
+        .sort((a, b) => {
+          const roleA = a.roles[0] || '';
+          const roleB = b.roles[0] || '';
+          return roleOrder[roleA] - roleOrder[roleB];
+        }) || [],
+    [projectContext.projectDataLoader.data.projectData.participants]
+  );
+
+  function getRandomHexColor(seed: number) {
```

Comment:
not necessary, but you could consider moving this into `utils/Utils.tsx`, which just has a collection of standalone functions.

Reply:
Moved.